Dropped anti drop items now lose the antidrop status if not initialized with it

farie82 opened the pull request:
**What does this PR do:**
If an item magically drops (or drops due to surgery) while it has nodrop it'll now get nodrop removed from it's flags if the item didn't have it when it was created.
There might be a few items that implement antidrop weirdly but those are usually admin spawned stuf.

Fixes: #7300

**Changelog:**
:cl:
fix: Items with NODROP that are dropped now lose the flag if they didn't get created with it
/:cl:

Fox-McCloud commented on code/game/objects/items.dm:
```diff
@@ -340,6 +340,8 @@ var/global/image/fire_overlay = image("icon" = 'icons/goonstation/effects/fire.d
 	if(flags & DROPDEL)
 		qdel(src)
+	if((flags & NODROP) && !(initial(flags) & NODROP)) //Remove NODROP is dropped
+		flags &= ~NODROP
```
this will probabbbbblyyy break things.

farie82 replied:
Most antidrop items either have it set in their construction. On pickup or by admemery. 
I know the last one will break.
If it's set when the person is holding it in his hands then it should be removed after it's gone from it's hands and reset once somebody picks it up.

It will break stuff but that stuff is probably poorly made. Fixing them would be cleaner than making more edge cases for antidrop (on surgery removal, on gib.. etc)

farie82 commented:
I made a list of the items using antidrop. ! means that it might break.

dark_priest (post_equip)
horsemask (magical * 2) (on cast) ! (curse removed on dead? Head removed = dead, unless IPC)
cling stuf (on init)
cult stuf (on init)
abductor vest (false on init, toggle) ! (How the hell will you remove something bound to the chest)
bloodcrawl (init)
sling stuf (init)
godshand (init)
doomslayer (mineEffect) (on effect) ! (2 minutes chainsaw. Will qdel once deactivated)
helmets (electropack having nodrop) (on attackby) ? (weird one. Shouldn't matter much
chrono_gun (init)
borg_defib (init (old))
nullrod items (init)
his grace (toggle)
cyborg ore bags (init)
owlman belt (init)
greytide toolbox (init)
water mister (init)
extinguisher nozzle (init)
syndi chainsaw (wield, removed on drop)
mutagen/hellwater cookie (on spawn) ! (despawn on dropped? DELDROP?)
all items on hotelguards (spawn, dusted)
cyborg glasses (init)
lasereyes (init)
godeye (init)
cluwne stuff (init)
shamebrero (init)
muzzle (toggle)
greytide gloves (init)
vox boots (toggle but removed on dropped)
chrono suit/helmet (activate) !
hardsuit attached helmet/boots (equip, removed on dropped) ! (refactor to do the same as the hoodie code)
rig_module/handheld (new)! (yeah this shit is odd. Same deal as hardsuit, unused)
rig (toggle)
rig clothing (init)
dodgeball (init)
greytide suit (init)
dogwhistle (on use for the doge)
cyborg stuff (init)
golems (init)
superhero items (on create) ! (if a superhero loses his clothing then this will happen)
ninja stuf (activate) ! (deactivate on drop)
finger gun (init)
cyberimp arm (extend)! (qdel on drop?)

Ralta commented:
Sweet! This has been an annoying little occurrence for a couple of years now - rare as it is.

farie82 commented:
If somebody else wants to take on the PR feel free.
